README.md

```diff
@@ -3,8 +3,76 @@
 ![GitHub release (latest by date)](https://img.shields.io/github/v/release/Liftric/octopus-deploy-plugin)
 [![CircleCI](https://circleci.com/gh/Liftric/octopus-deploy-plugin/tree/master.svg?style=svg)](https://circleci.com/gh/Liftric/octopus-deploy-plugin/tree/master)
 
+The octopus-deploy-plugin generates build-information and uploads packages and generated build-information to Octopus Deploy instances.
+
+Requirements:
+ * [Octopus Deploy CLI](https://octopus.com/downloads/octopuscli)
+ * git (for commit/changelog calculation)
+ 
 # Usage
+The plugin can be configured with the `octopus` DSL block:
+```
+plugins {
+    id("com.liftric.octopus-deploy-plugin") version "whatever"
+}
+[...]
+octopus {
+    serverUrl = "http://localhost:8080/"
+    apiKey = "API-TESTTEST123TRESDTSDD"
+
+    generateChangelogSinceLastTag = true
+
+    val jar by tasks.existing(Jar::class)
+    packageName = jar.get().archiveBaseName.get()
+    version = jar.get().archiveVersion.get()
+    pushPackage = jar.get().archiveFile.get().asFile
+}
+```
+
+See the example directory for a minimal, working example.
+
+## tasks
+The following tasks are added by the plugin:
+
+Task | Description
+---|---
+previousTag | Calls git describe to receive the previous tag name. Will fail if no tag is found.
+firstCommitHash | Calls git log to get the first commit hash of the current history tree
+commitsSinceLastTag | Calls git log to receive all commits since the previous tag or the first commit of the current history.
+createBuildInformation | Creates the octopus build-information file.
+uploadBuildInformation | Uploads the created octopus build-information file.
+uploadPackage | Uploads the package to octopus.
+
+For normale use-cases, only `uploadBuildInformation` are `uploadPackage` are needed to call explicitly. Depending
+task will be called implicitly by both as needed.
+
+**Noteworthy**: The build-information can be uploaded before the package itself. 
+Useful when creating automatic releases and using the commits in the release notes in octopus.
+
+## naming
+Octopus deploy expects the name and version in the following format: `<name>.<version>.<extension>`
+
+For Java archives, `<name>-<version>.<extension>` is conventional, so should be changed to get picked up properly by octopus.
+Otherwise the first version number part will be parsed as part of the name.
 
 ## configuration
+After the plugin is applied, the vault extension is registered with the following settings:
```

Comment:
I guess a typo happened: `vault` to `octopus deploy` extension?

Reply:
More like copy paste error 😄